### PR TITLE
Remove irrelevant suggestion of a legacy method from a warning

### PR DESCRIPTION
**Pull request by zx6658**
As react version is updated and some lifecycle is deprecated, warn message should be changed like in my pr :)

**Warning Content:**
Cannot update during an existing state transition (such as withinrenderor another component's constructor). Render methods should be a pure function of props and state; constructor side-effects are an anti-pattern, but can be moved to **componentWillMount.**

**What is the expected behavior?**
I thought, warning messages what use deprecated lifecycle method should be changed!! :)

### CODE REVIEW

File: packages/react-dom/src/__tests__/ReactCompositeComponent-test.js [hunk 1]
```diff
@@ -432,8 +432,8 @@ describe('ReactCompositeComponent', () => {
     }).toWarnDev(
       'Cannot update during an existing state transition (such as within ' +
         "`render` or another component's constructor). Render methods should " +
-        'be a pure function of props and state; constructor side-effects are ' +
-        'an anti-pattern, but can be moved to `componentWillMount`.',
+        'be a pure function of props and state. ' +
+        'Move the setState call to componentDidMount`.',
```
Comment (gaearon) [inline packages/react-dom/src/__tests__/ReactCompositeComponent-test.js]: This is too prescriptive. We don't actually *know* they called setState in constructor, or that they want to move it to cDM (setting it in constructor might be better in most cases).

Comment (zx6658) [reply packages/react-dom/src/__tests__/ReactCompositeComponent-test.js]: @gaearon I agree:) Whatever, i thought deprecated lifecycle error message should be deleted! 
As you say, I will fix pr by setting warn message by deleting message about constructor and cDM

Comment (zx6658) [reply packages/react-dom/src/__tests__/ReactCompositeComponent-test.js]: @gaearon is it ok?

**Comment by FancyCoder0** [comment]
Hi @zx6658 ,
We are researchers working on identifying redundant development and duplicated pull requests. We would like to help open source community to reduce redundant work. We have found there is another open pull request: https://github.com/facebook/react/pull/12760 which might be a potentially duplicate to this one. We would really appreciate if you could help us to validate and give us feedback.

**Pull request merged**